x-pack/legacy/plugins/apm/index.ts

```diff
@@ -17,7 +17,14 @@ import { plugin } from './server/new-platform/index';
 
 export const apm: LegacyPluginInitializer = kibana => {
   return new kibana.Plugin({
-    require: ['kibana', 'elasticsearch', 'xpack_main', 'apm_oss'],
+    require: [
+      'kibana',
+      'elasticsearch',
+      'xpack_main',
+      'apm_oss',
+      'alerting',
```

Comment:
I'm not sure if these should be required, or if they are optional. If it's the latter, not sure how we can get access to these plugins on startup.

Reply:
Really good question for platform how to handle optional dependencies. I think it has come up before but can't remember the answer.

Reply:
New platform allows for optional dependencies, but legacy platform does not.  If you are depending on something in a legacy plugin that could be disabled, you need to make sure that you implement isEnabled and check that the dependency is enabled, otherwise Kibana will crash.